xarray/core/dtypes.py

```diff
@@ -78,7 +78,9 @@ def maybe_promote(dtype):
     fill_value : Valid missing value for the promoted dtype.
     """
     # N.B. these casting rules should match pandas
-    if np.issubdtype(dtype, np.floating):
+    if np.issubdtype(dtype, np.timedelta64):
```

Comment:
can you add a comment with a link to the numpy bug explaining why we need to do this first?

Reply:
Also maybe move this below np.floating -- there was an intention to hit more commonly used dtypes first.